contracts/script-utils/src/class.rs

```diff
@@ -1 +1,93 @@
-pub const CLASS_TYPE_ARGS_LEN: usize = 52;
+use crate::error::Error;
+use crate::helper::parse_dyn_vec_len;
+use alloc::vec::Vec;
+use core::result::Result;
+
+const CLASS_DATA_MIN_LEN: usize = 16;
+pub const CLASS_TYPE_ARGS_LEN: usize = 24;
+
+/// Class cell data structure
+/// This structure contains the following information:
+/// 1) version: u8
+/// 2) total: u32
+/// 3) issued: u32
+/// 4) configure: u8
+/// 5) name: <size: u16> + <content>
+/// 6) description: <size: u16> + <content>
+/// 7) renderer: <size: u16> + <content>
+/// 8) extinfo_data: <size: u16> + <content>
+/// The fields of 1), 2), 4), 5) and 6) cannot be changed after they are set and they cannot be
+/// missing. The fields of 3) and 7) can be changed and it cannot be missing.
+/// The filed of 8) can be changed and it also can be missing and it will not be validated.
+pub struct Class {
+    pub version:     u8,
+    pub total:       u32,
+    pub issued:      u32,
+    pub configure:   u8,
+    pub name:        Vec<u8>,
+    pub description: Vec<u8>,
+}
+
+impl Class {
+    pub fn from_data(data: &[u8]) -> Result<Self, Error> {
+        if data.len() < CLASS_DATA_MIN_LEN {
+            return Err(Error::ClassDataInvalid);
+        }
+
+        let version: u8 = data[0];
+        if version != 0 {
+            return Err(Error::VersionInvalid);
+        }
+
+        let mut total_buf = [0u8; 4];
+        let mut issued_buf = [0u8; 4];
+        total_buf.copy_from_slice(&data[1..5]);
+        issued_buf.copy_from_slice(&data[5..9]);
+        let total = u32::from_be_bytes(total_buf);
+        let issued = u32::from_be_bytes(issued_buf);
```

Comment:
Can we write a helper function to read buf and convert type?

Reply:
Good idea!

Reply:
Updated. https://github.com/nervina-labs/ckb-nft-scripts/pull/2/commits/1f00af215deed929755387ccb489754f1b01398a